powerline.go

```diff
@@ -176,8 +186,19 @@ func (p *powerline) draw() string {
 		buffer.WriteString(p.reset)
 	}
 	buffer.WriteRune(' ')
+	return buffer.String()
+}
+
+func (p *powerline) draw() string {
+
+	rows := make([]string, 0)
```

Comment:
This implementation does lots of costly allocations during drawing, slowing down powerline-go quite a bit – can you draw the rows directly?

Reply:
How about now? It only allocates a single bytes.Buffer now, and only creates a the prompt string once, at the very end.

(Still learning how to navigate Github pull requests, but I think the updates I made should be visible...)